File: scripts/generate-test-data.js
const fs = require('fs');
const path = require('path');

const products = [
    { name: "Latte", price: 4.50, category: "Coffee" },
    { name: "Espresso", price: 3.00, category: "Coffee" },
    { name: "Cappuccino", price: 4.00, category: "Coffee" },
    { name: "Americano", price: 3.50, category: "Coffee" },
    { name: "Oat Milk Latte", price: 5.50, category: "Coffee" },
    { name: "Croissant", price: 3.75, category: "Pastry" },
    { name: "Blueberry Muffin", price: 3.50, category: "Pastry" },
    { name: "Avocado Toast", price: 9.00, category: "Food" },
    { name: "Breakfast Sandwich", price: 8.50, category: "Food" },
    { name: "Hot Tea", price: 3.00, category: "Tea" },
];

const paymentMethods = ["Credit Card", "Digital Wallet", "Cash"];
const baseDate = new Date(2026, 1, 8); // February (0-indexed so 1 is Feb)
const testDataDir = path.join(process.cwd(), 'test-data');

if (!fs.existsSync(testDataDir)) {
    fs.mkdirSync(testDataDir, { recursive: true });
}

for (let i = 0; i < 5; i++) {
    const currentDate = new Date(baseDate);
    currentDate.setDate(baseDate.getDate() + i);

    const dateStr = currentDate.toISOString().split('T')[0];
    const filename = path.join(testDataDir, `kaapi_ledger_${dateStr.replace(/-/g, '_')}.csv`);
    const numTransactions = Math.floor(Math.random() * (300 - 100 + 1)) + 100;

    let csvContent = "Transaction ID,Date,Time,Product,Category,Qty,Price,Total,Payment Method\n";

    for (let j = 0; j < numTransactions; j++) {
        const productObj = products[Math.floor(Math.random() * products.length)];
        const qty = [1, 1, 1, 1, 2, 2, 3][Math.floor(Math.random() * 7)];
        const total = (productObj.price * qty).toFixed(2);

        const hour = Math.floor(Math.random() * (18 - 7 + 1)) + 7;
        const minute = Math.floor(Math.random() * 60);
        const timeStr = `${hour.toString().padStart(2, '0')}:${minute.toString().padStart(2, '0')}`;

        csvContent += `TXN-${dateStr.replace(/-/g, '')}-${j.toString().padStart(3, '0')},${dateStr},${timeStr},${productObj.name},${productObj.category},${qty},${productObj.price.toFixed(2)},${total},${paymentMethods[Math.floor(Math.random() * paymentMethods.length)]}\n`;
    }

    fs.writeFileSync(filename, csvContent);
    console.log(`Generated ${filename} with ${numTransactions} transactions.`);
}

File: scripts/generate_test_data.py
import csv
import random
from datetime import datetime, timedelta

# Configuration
products = [
    ("Latte", 4.50, "Coffee"),
    ("Espresso", 3.00, "Coffee"),
    ("Cappuccino", 4.00, "Coffee"),
    ("Americano", 3.50, "Coffee"),
    ("Oat Milk Latte", 5.50, "Coffee"),
    ("Croissant", 3.75, "Pastry"),
    ("Blueberry Muffin", 3.50, "Pastry"),
    ("Avocado Toast", 9.00, "Food"),
    ("Breakfast Sandwich", 8.50, "Food"),
    ("Hot Tea", 3.00, "Tea"),
]

payment_methods = ["Credit Card", "Digital Wallet", "Cash"]
base_date = datetime(2026, 2, 8)

for i in range(5):
    current_date = base_date + timedelta(days=i)
    filename = f"test-data/kaapi_ledger_{current_date.strftime('%Y_%m_%d')}.csv"
    num_transactions = random.randint(100, 300)
    
    with open(filename, 'w', newline='') as f:
        writer = csv.writer(f)
        writer.writerow(["Transaction ID", "Date", "Time", "Product", "Category", "Qty", "Price", "Total", "Payment Method"])
        
        for j in range(num_transactions):
            product, price, category = random.choice(products)
            qty = random.choices([1, 2, 3], weights=[80, 15, 5])[0]
            total = round(price * qty, 2)
            
            # Spread transactions throughout the day (7 AM to 7 PM)
            hour = random.randint(7, 18)
            minute = random.randint(0, 59)
            time_str = f"{hour:02d}:{minute:02d}"
            
            writer.writerow([
                f"TXN-{current_date.strftime('%Y%m%d')}-{j:03d}",
                current_date.strftime('%Y-%m-%d'),
                time_str,
                product,
                category,
                qty,
                f"{price:.2f}",
                f"{total:.2f}",
                random.choice(payment_methods)
            ])
            
    print(f"Generated {filename} with {num_transactions} transactions.")
